docs/node/migrations/zero-downtime-migrations/state-replication.mdx

```diff
@@ -0,0 +1,68 @@
+---
+sidebar_position: 2
+sidebar_label: "State Replication"
+id: state-replication
+title: State Replication
+description: State Replication
+slug: /node/migrations/zero-downtime-migrations/state-replication
+---
+
+Zero-Downtime Migrations require the new network to start with the state of the existing network at the start of the migration and then replicate the state changes of the existing network to the new network during the migration period to stay in sync with the existing network.
+
+## State Replication
+
+State replication is handled by an inbuilt `migrations` listener extension in the Kwil node. The `migrations` listener extension polls for the state changes on the existing network and replicates the changes onto the new network. The `migrations` listener extension can be enabled by setting the below configuration in the `config.toml` and `genesis.json` files.
+
+
+`migrate_from` config in the `config.toml` file specifies the JSON-RPC address of the node from the existing network that the new network should replicate the state changes from. 
+```toml
+[app]
+"migrate_from" = "http://kwild-jsonrpc-server-address:8484"
+```
+
+The `consensus_params` in the `genesis.json` file specify the start and end block heights that the new network should use to replicate the state changes from the existing network.
+```json
+{
+    "consensus_params": {
+        "migration": {
+            "start_height": <start_height>,
+            "end_height": <end_height>,
+        },
+    }
+}
+
+```json
+{
+    "consensus_params": {
+        "migration": {
+            "start_height": <start_height>,
+            "end_height": <end_height>,
+        },
+    }
+}
+```
+:::note 
+Migrations listener extension won't start if the `start_height` and `end_height` are not set in the `genesis.json` file.
+:::
+
+## Handling State Divergence
+
+During migration period, as both old and new networks are running concurrently handing user transactions and replicating state changes from the existing network, there is a possibility for the occurance of state divergences between the two networks. 
+The migration process does its best to resolve the conflicts between the two networks using the below conflict resolution principles. However, there might be some schemas where the conflicts cannot be resolved without any loss of data.
```

Comment:
"The below" feels a bit awkward. I would just say that there are conflict resolution principles and then list them below.

```suggestion
The migration process does its best to resolve the conflicts between the two networks using a set of conflict resolution principles. However, there might be some schemas where the conflicts cannot be resolved without any loss of data.
```